src/smoothscroll.js

```diff
@@ -1,371 +1,436 @@
+        this,
+        arguments[0] === undefined
+          ? true
+          : arguments[0]
       );
-    };
 
-    // Element.prototype.scrollBy
-    Element.prototype.scrollBy = function() {
-      var arg0 = arguments[0];
+      return;
+    }
+
+    // LET THE SMOOTHNESS BEGIN!
+    var scrollableParent = findScrollableParent(this);
+    var parentRects = scrollableParent.getBoundingClientRect();
+    var clientRects = this.getBoundingClientRect();
 
-      if (typeof arg0 === 'object') {
-        this.scroll({
-          left: arg0.left + this.scrollLeft,
-          top: arg0.top + this.scrollTop,
-          behavior: arg0.behavior
-        });
-      } else {
-        this.scroll(
-          this.scrollLeft + arg0,
-          this.scrollTop + arguments[1]
-        );
-      }
-    };
-
-    // Element.prototype.scrollIntoView
-    Element.prototype.scrollIntoView = function() {
-      // avoid smooth behavior if not required
-      if (shouldBailOut(arguments[0])) {
-        original.scrollIntoView.call(
-          this,
-          arguments[0] === undefined ? true : arguments[0]
-        );
-        return;
-      }
+    if (scrollableParent !== d.body) {
+      // reveal element inside parent
+      smoothScroll.call(
+        this,
+        scrollableParent,
+        scrollableParent.scrollLeft + clientRects.left - parentRects.left,
+        scrollableParent.scrollTop + clientRects.top - parentRects.top
+      );
 
-      // LET THE SMOOTHNESS BEGIN!
-      var scrollableParent = findScrollableParent(this);
-      var parentRects = scrollableParent.getBoundingClientRect();
-      var clientRects = this.getBoundingClientRect();
-
-      if (scrollableParent !== d.body) {
-        // reveal element inside parent
-        smoothScroll.call(
-          this,
-          scrollableParent,
-          scrollableParent.scrollLeft + clientRects.left - parentRects.left,
-          scrollableParent.scrollTop + clientRects.top - parentRects.top
-        );
-        // reveal parent in viewport
+      // reveal parent in viewport unless is fixed
+      if (w.getComputedStyle(scrollableParent).position !== 'fixed') {
         w.scrollBy({
           left: parentRects.left,
           top: parentRects.top,
           behavior: 'smooth'
         });
-      } else {
-        // reveal element in viewport
-        w.scrollBy({
-          left: clientRects.left,
-          top: clientRects.top,
-          behavior: 'smooth'
-        });
       }
-    };
-  }
-
-  if (typeof exports === 'object') {
-    // commonjs
-    module.exports = { polyfill: polyfill };
-  } else {
-    // global
-    polyfill();
-  }
-})(window, document);
+    } else {
+      // reveal element in viewport
+      w.scrollBy({
+        left: clientRects.left,
+        top: clientRects.top,
+        behavior: 'smooth'
+      });
+    }
+  };
+}
+
+if (typeof exports === 'object') {
+  // commonjs
+  module.exports = { polyfill: polyfill };
+} else {
+  // global
+  polyfill();
```

Comment:
it is a bit weird to me that based on how you import the file you get different behavior, but this is how we currently do it so let’s not change it...yet.

Also, does rollup help us much in this project since we’re not using es6 modules (or any modules for that matter)?

Reply:
I know there was a reason why we wanted a function to be called for the module, to be honest I don't remember it right now. Let's keep it for now, on a next release we can discuss a new approach.

About rollup, it doesn't do _much_ right now. Main thing, it prevents us to have an IIFE wrapping all the code, adding 2 more unnecessary spaces per line. Also it handles the release banner on top of the dist file.

But the most important part is to have the ground prepare for future transformations on the code, like adding typing, babel plugins, etc.